ui/pages/Project/components/RnaSeqResultPage.jsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Loader, Grid, Dropdown } from 'semantic-ui-react'
+
+import { getGenesById, getIndividualsByGuid, getRnaSeqDataByIndividual } from 'redux/selectors'
+import { RNASEQ_JUNCTION_PADDING, TISSUE_DISPLAY } from 'shared/utils/constants'
+import DataLoader from 'shared/components/DataLoader'
+import { loadRnaSeqData } from '../reducers'
+import { getRnaSeqDataLoading, getRnaSeqSignificantJunctionData } from '../selectors'
+
+const RnaSeqOutliers = React.lazy(() => import('./RnaSeqOutliers'))
+const RnaSeqOutliersTable = React.lazy(() => import('./RnaSeqJunctionOutliersTable'))
+
+const OUTLIER_VOLCANO_PLOT_CONFIGS = {
+  outliers: {
+    getLocation: (({ geneId }) => geneId),
+    searchType: 'genes',
+    title: 'Expression Outliers',
+    formatData: ({ data }) => data,
+  },
+  spliceOutliers: {
+    getLocation: (({ chrom, start, end }) => `${chrom}:${Math.max(1, start - RNASEQ_JUNCTION_PADDING)}-${end + RNASEQ_JUNCTION_PADDING}`),
+    searchType: 'regions',
+    title: 'Splice Junction Outliers',
+    formatData: ({ data, tissueType }) => data.filter(outlier => outlier.tissueType === tissueType),
+  },
+}
+
+class BaseRnaSeqResultPage extends React.PureComponent {
+
+  static propTypes = {
+    individual: PropTypes.object,
+    rnaSeqData: PropTypes.object,
+    significantJunctionOutliers: PropTypes.arrayOf(PropTypes.object),
+    genesById: PropTypes.object,
+  }
+
+  state = {
+    tissueType: null,
+    tissueOptions: null,
+  }
+
+  componentDidMount() {
+    const { rnaSeqData } = this.props
+    const tissueTypes = Array.from(Object.values(rnaSeqData?.spliceOutliers || {}).flat().reduce(
+      (acc, { tissueType }) => acc.add(tissueType), new Set(),
+    ))
+    const tissueOptions = tissueTypes.map(tissueType => (
+      { key: tissueType, text: TISSUE_DISPLAY[tissueType] || 'No Tissue', value: tissueType }
+    ))
+    if (tissueOptions.length) {
+      this.setState({ tissueType: tissueOptions[0].value, tissueOptions })
```

Comment:
`tissueOptions` should a be a prop calculated by a selector, and then you should set the intial state based on the props in the constructor
```
    constructor(props) {
        super(props)
        this.state = { tissueType: props.tissueOptions && props.tissueOptions[0] }
    }
```

Reply:
We have a lint rule: no react/state-in-constructor. Do you think it is okay to suppress it?

Reply:
yes we disable it in at least 3 other places so thats okay